py/server/deephaven/time.py

```diff
@@ -1709,18 +1732,14 @@ def lower_bin(dt: Union[Instant, ZonedDateTime], interval: Union[int, str], offs
         DHError
     """
     try:
-        if isinstance(interval, str):
-            interval = parse_duration_nanos(interval)
-
-        if isinstance(offset, str):
-            offset = parse_duration_nanos(offset)
-
```

Comment:
Here we've traded branching for rewriting the whole API with `Object`.  This is a problem.

Reply:
Thinking more about the branching problem, I think a big part of it is a dispatch problem.  We aren't using the type information from the columns or input variables to execute the correct function.  Instead, it is all getting forced into this object casting stuff, which I have big concerns about.